src/main/java/de/tum/in/www1/artemis/service/connectors/vcs/AbstractVersionControlService.java

```diff
@@ -163,6 +164,9 @@ public String getOrRetrieveBranchOfStudentParticipation(ProgrammingExerciseStude
     @Override
     public String getOrRetrieveBranchOfExercise(ProgrammingExercise programmingExercise) {
         if (programmingExercise.getBranch() == null) {
+            if (!Hibernate.isInitialized(programmingExercise.getTemplateParticipation())) {
+                programmingExercise = programmingExerciseRepository.findWithTemplateAndSolutionParticipationByIdElseThrow(programmingExercise.getId());
```

Comment:
How about, instead of reloading the whole exercise, we only update the section that we tested for?
=> Retrieve the template participation of the exercise and set it, if it's needed further down.